stl/inc/xutility

```diff
@@ -4187,6 +4187,313 @@ struct default_sentinel_t {};
+        requires random_access_iterator<_Iter>
+    {
+#if _ITERATOR_DEBUG_LEVEL != 0
+        _STL_VERIFY(-_Diff <= _Length, "counted_iterator decrement beyond end of range");
+        if constexpr (_Offset_verifiable_v<_Iter>) {
+            _Current._Verify_offset(-_Diff);
+        }
+#endif // _ITERATOR_DEBUG_LEVEL != 0
+        return counted_iterator(_Current - _Diff, _Length + _Diff);
+    }
+    // clang-format on
+
+    template <common_with<_Iter> _Other>
+    friend constexpr iter_difference_t<_Other> operator-(
+        const counted_iterator& _Left, const counted_iterator<_Other>& _Right) {
+        _Verify_Range(_Left, _Right);
+        return _Right._Length - _Left._Length;
+    }
+    friend constexpr iter_difference_t<_Iter> operator-(const counted_iterator& _Left, default_sentinel_t) {
+        return -_Left._Length;
+    }
+    friend constexpr iter_difference_t<_Iter> operator-(default_sentinel_t, const counted_iterator& _Right) {
+        return _Right._Length;
+    }
+
+    // clang-format off
+    constexpr counted_iterator& operator-=(iter_difference_t<_Iter> _Diff)
+        requires random_access_iterator<_Iter>
+    {
+#if _ITERATOR_DEBUG_LEVEL != 0
+        _STL_VERIFY(-_Diff <= _Length, "counted_iterator decrement beyond end of range");
+        if constexpr (_Offset_verifiable_v<_Iter>) {
+            _Current._Verify_offset(-_Diff);
+        }
+#endif // _ITERATOR_DEBUG_LEVEL != 0
+        _Current -= _Diff;
+        _Length += _Diff;
+        return *this;
+    }
+    // clang-format on
+
+    // [counted.iter.cmp]
+    template <common_with<_Iter> _Other>
+    friend constexpr bool operator==(const counted_iterator& _Left, const counted_iterator<_Other>& _Right) {
+        _Verify_Range(_Left, _Right);
+        return _Left._Length == _Right._Length;
+    }
+
+    friend constexpr bool operator==(const counted_iterator& _Left, default_sentinel_t) noexcept // strengthened
+    {
+        return _Left._Length == 0;
+    }
+
+    template <common_with<_Iter> _Other>
+    friend constexpr strong_ordering operator<=>(
+        const counted_iterator& _Left, const counted_iterator<_Other>& _Right) {
+        _Verify_Range(_Left, _Right);
+        return _Right._Length <=> _Left._Length;
+    }
+
+    // [counted.iter.cust]
+    // clang-format off
+    friend constexpr iter_rvalue_reference_t<_Iter> iter_move(const counted_iterator& _It) noexcept(
+        noexcept(_RANGES iter_move(_It.current)))
+       requires input_iterator<_Iter>
+    {
+        return _RANGES iter_move(_It.current);
+    }
+    // clang-format on
+
+    template <indirectly_swappable<_Iter> _Other>
+    friend constexpr void iter_swap(const counted_iterator& _Left, const counted_iterator<_Other>& _Right) noexcept(
+        noexcept(_RANGES iter_swap(_Left._Current, _Right.base()))) {
+        _RANGES iter_swap(_Left._Current, _Right.base());
+    }
+
+    template <class _Other, enable_if_t<_Range_verifiable_v<_Iter, _Other>, int> = 0>

```

Comment:
Ah! `enable_if_t` in concepts code! It burns our eyes! (Several places.)

Reply:
I does so. But I was unsure whether this is one of the places where it would be ill formed similar to the one occurence in view_interface.

That said 🔥